### ISSUE TITLE
[OBS CRASH] with NVIDIA AR MOVE

### ISSUE THREAD

**Issue by DaDiGui** [exeldro/obs-move-transition#189]
I am trying to follow this [tutorial](https://discord.com/channels/157891598867234817/1158457162884984893) but obs is crashing every time I set action 1.

https://obsproject.com/logs/rJaABOVG0CfOcBLT
`14:36:08.649: [Move Transition] Error in Run landmark; error -1: An otherwise unspecified error has occurred`

**Comment by DaDiGui** [comment]
log: https://obsproject.com/logs/Fk0YvBVtOOGwYz3J

**Comment by exeldro** [comment]
It is an error I have no experience in solving, what I notice in the log file is:
`[Move Transition] Error in Load expression; error -30: There was a problem deserializing the inference runtime engine`
I tried the same steps on my RTX 2060 without a problem

**Comment by shuffgy** [comment]
i got the same error and crashes if i want to try landmark or body tracking it is instantly crashing.

i installed the today current AR SDK and the today current move plugin version.

[crashlog](https://obsproject.com/logs/oKYrGRyaW7Y9XpmM), [normal log](https://obsproject.com/logs/z62RaiVjcrNgyixC)

i also noticed i have two duplicated inputs, maybe thats one of the reasons @exeldro ?

![Screenshot 2024-09-21 034130](https://github.com/user-attachments/assets/107a910f-2a3f-46c6-8fd2-3218f0ddbbd7)

**Comment by funbasealpha** [comment]
I'm also getting this instacrash and would love love if i wasn't.
[crash log](https://obsproject.com/logs/dQ7PggRYvkYjo8K4)

**Comment by funbasealpha** [comment]
I'm also getting this crash [log](https://obsproject.com/logs/QdIgo3UcL83B0i2K)

**Comment by exeldro** [comment]
@funbasealpha that OBS log file has a successful shutdown, if you have a crash please provide the OBS log file of the session with the issue and provide the OBS crash report.

**Comment by funbasealpha** [comment]
oh my mistake, after an OBS update i can now set it to landmark, but it doesnt seem to function. switching to body crashed it. [log from that one](https://obsproject.com/logs/FVY2eiW4sN5rVksy)

**Comment by funbasealpha** [comment]
[Crash 2025-01-26 15-23-38.txt](https://github.com/user-attachments/files/18552037/Crash.2025-01-26.15-23-38.txt) [log file](https://obsproject.com/logs/OdsN1NVPawAD7rd9) One last try. I don't really know what I'm doing here as I'm sure you've gathered by now.

**Comment by dighawaii** [comment]
1 year later and I think it's safe to say this plugin is deprecated.  Freezes OBS whenever switching to a scene that has an AR Move filter in it.

**Comment by dighawaii** [comment]
Fair. I suppose it's still working for many. I uninstalled it and will try a reinstall later. It was causing obs to freeze, even though I have not been using the plugin. If you want to use my pc as a test bed, you're welcome to it, full access. Thanks for all the work you do!

**Comment by LetsDoVideo** [comment]
Hi Exeldro. It's me BoomSoMuch! I had this exact issue a year ago when I reinstalled Windows. I kept trying different versions of the AR SDK and NVIDIA drivers. I eventually got it working and can't remember the fix because I am an idiot.

I just reinstalled Windows again and having the same issue. If there is anything I can do to help, give you access to my machine, run tests, get logs, anything! I'm not worried about using it myself, but you've done so much to help me over the years, anything I can do just let me know. 

I wish I could give you better steps to reproduce it. I wiped my drive, reinstalled Win 10 from a USB stick. Installed all my stuff, dragged my portables back over to where they need to be. Everything works except this one filter.

<img width="585" height="227" alt="Image" src="https://github.com/user-attachments/assets/86e8e8e2-be62-4053-b5fe-b7622dfb9877" />

[Crash 2026-04-21 13-45-08.txt](https://github.com/user-attachments/files/26942450/Crash.2026-04-21.13-45-08.txt)

[Crash 2026-04-21 12-59-24.txt](https://github.com/user-attachments/files/26942451/Crash.2026-04-21.12-59-24.txt)

[2026-04-21 13-44-02.txt](https://github.com/user-attachments/files/26942459/2026-04-21.13-44-02.txt)

[2026-04-21 13-35-08.txt](https://github.com/user-attachments/files/26942460/2026-04-21.13-35-08.txt)

**Comment by LetsDoVideo** [comment]
**Update: Resolved!**

Finally got it working. Sharing the fix in case it helps you or anyone else running into this.

**The working combination:**

- NVIDIA AR SDK **0.8.7** (not 0.8.2)
- Move Transition plugin **3.1.4**
- NVIDIA **Game Ready** driver (current — I'm on an RTX 2060 Super / Windows 10)

**The root cause as best I can tell:**

On a fresh Windows install, AR SDK 0.8.2 refuses to deserialize its pre-compiled TensorRT engines against newer NVIDIA drivers and fails with "error -30: There was a problem deserializing the inference runtime engine" and crashes OBS when the filter runs against actual video.

AR SDK 0.8.7 has newer engines that work with current drivers.

Note on drivers: I was initially on the Studio driver and ran into issues. Switching to Game Ready was part of the sequence that eventually got it working, though I can't say for certain it was strictly required.

**The tricky part that confused me:**

My existing portable OBS installs had **Move Transition 3.1.1**. When I installed AR SDK 0.8.7, Move 3.1.1 detected it (the log showed `NVIDIA AR version: 0.8.7.0`) but silently didn't register the `nv_move_filter` source type. So the plugin loaded, but the AR filter type was unavailable.  Existing filter instances in my scenes had no properties, and I couldn't add new ones.

Upgrading Move Transition to 3.1.4 in those portables fixed it. 3.1.4 registers the filter properly with AR SDK 0.8.7 present.

**TL;DR for anyone else stuck on this:**

If you're getting "deserializing the inference runtime engine" errors with AR SDK 0.8.2, try AR SDK 0.8.7 instead. And make sure your Move Transition plugin is **at least 3.1.4**. Older versions don't play well with 0.8.7.

Thanks for the amazing plugin, and happy to help debug in the future if you hit reports like this.
